joomla/joomla-cms
# [4.0][CLI] Add a task to discover extensions

### Summary of Changes

i added a CLI task to discover extensions. I took https://github.com/joomla/joomla-cms/pull/28666 as an example


### Testing Instructions
1. Apply this PR
2. Copy or symlink an extension to your Joomla installation
3. Run ~~` php ./cli/joomla.php extension:discover --eid=122`~~ ` php ./cli/joomla.php extension:discoverinstall --eid=122` for discovering the extension with the ID 122 
or 
Run the command without a parameter (like this ~~`php ./cli/joomla.php extension:discover`~~ `php ./cli/joomla.php extension:discoverinstall`) for discovering all extensions, that are copied to Joomla but not installed.

4. Run `php ./cli/joomla.php list` or only `php ./cli/joomla.php` and check the entry in the list.
5. Run ~~`php ./cli/joomla.php help extension:discover`~~ `php ./cli/joomla.php help extension:discoverinstall`to check the help information.



### Documentation Changes Required
I would add it here: https://docs.joomla.org/J4.x:CLI_Update
